### ISSUE TITLE
Pulling containers is broken (dial tcp [IPv6]:443: connect: network is unreachable)

### ISSUE THREAD

**Issue by MidnightSlicer** [spantaleev/matrix-docker-ansible-deploy#2860]
**Describe the bug**
Docker changed the way their stuff works on the backend and it seems to have broke this project's ability to pull new containers. 

It seems like `docker pull` cannot talk to Docker's servers anymore. Here is the exact error message I am getting:
```
fatal: [matrix.<domain.tld>]: FAILED! => changed=false 
  attempts: 10
  msg: 'Error pulling image docker.io/devture/exim-relay:4.96-r1-0 - 500 Server Error for http+docker://localhost/v1.43/images/create?tag=4.96-r1-0&fromImage=docker.io%2Fdevture%2Fexim-relay: Internal Server Error ("Get "https://registry-1.docker.io/v2/devture/exim-relay/manifests/sha256:6aeb52edd282561aeaaa41de5757ba714419b21eb3e1454d9658d2bd0313fe14": dial tcp [2600:1f18:2148:bc02:4bf0:3a98:55ae:e3d5]:443: connect: network is unreachable")'
```

This thread covers it: https://forums.docker.com/t/unable-to-pull-docker-image-due-to-ipv6-since-2023-08-23-around-6pm-cest/137378/8

**To Reproduce**
Try to pull any container on a machine where Docker has been installed by this Ansible deploy. This can be done by a change in `vars.yml` or by just running `sudo docker pull <container>`.

**Expected behavior**
The container should be pulled.

**Matrix Server:**
 - OS: Debian 12
 - Architecture: amd64

**Additional context**
I am on a brand new install of Debian and attempting to migrate my server to it. I can't run the set up command because it can't pull all the containers it needs. I do not fully understand how this Ansible deploy works, but I did have it install Docker on this server because it doesn't seem to work if Docker was installed by anything else. I hope the information in that [Docker thread](https://forums.docker.com/t/unable-to-pull-docker-image-due-to-ipv6-since-2023-08-23-around-6pm-cest/137378/8) will help.

**Comment by spantaleev** [comment]
This only seems to be affecting you.

Try to search for "Linux disable IPv6" and follow any docs you find until you've killed IPv6 off completely.

**Comment by MidnightSlicer** [comment]
> This only seems to be affecting you.
> 
> Try to search for "Linux disable IPv6" and follow any docs you find until you've killed IPv6 off completely.

I did that and rebooted the server for good measure. That just changes the error to `socket: address family not supported by protocol`. 

I am honestly at a loss. After undoing the changes it now says `connect: network is unreachable`. I don't know if that was something that changed between last night and now or if I didn't fully undo everything. The only source for yesterday's error code (`connect: network is unreachable`) was this [thread](https://forums.docker.com/t/unable-to-pull-docker-image-due-to-ipv6-since-2023-08-23-around-6pm-cest/137378/8). The only other thing I have found that may be relevant is this issue tracker: https://www.dockerstatus.com/pages/incident/533c6539221ae15e3f000031/64e75cde1670c8052e6ffd6d

Should I just wait a few weeks and try again?
